tests/utils_tests/test_conv.py

```diff
@@ -0,0 +1,122 @@
+import unittest
+
+import numpy
+
+from chainer import cuda
+from chainer import testing
+from chainer.testing import attr
+from chainer.utils import conv
+
+
+if cuda.available:
+    cuda.init()
+
+
+class TestConv(unittest.TestCase):
+
+    def check_conv_outsize(self, size, k, s, p):
+        # When cover_all == False, `outsize` is the maximum integer that
+        # satisfies "(outsize - 1) * s + k <= w"
+        w = size + p * 2
+        outsize = conv.get_conv_outsize(size, k, s, p, cover_all=False)
+        self.assertTrue((outsize - 1) * s + k <= w < outsize * s + k)
+
+    def check_conv_outsize_cover_all(self, size, k, s, p):
+        # When cover_all == True, `outsize` is the minimum integer that
+        # satisfies "w <= (outsize - 1) * s + k"
+        w = size + p * 2
+        outsize = conv.get_conv_outsize(size, k, s, p, cover_all=True)
+        self.assertTrue((outsize - 2) * s + k < w <= (outsize - 1) * s + k)
+
+    def test_conv_outsize1(self):
+        self.check_conv_outsize(10, 4, 3, 2)
+
+    def test_conv_outsize2(self):
+        self.check_conv_outsize(10, 4, 4, 2)
+
+    def test_conv_outsize_cover_all1(self):
+        self.check_conv_outsize_cover_all(10, 4, 3, 2)
+
+    def test_conv_outsize_cover_all2(self):
+        self.check_conv_outsize_cover_all(10, 4, 4, 2)
+
+
+class TestIm2Col(unittest.TestCase):
+
+    def setUp(self):
+        self.img = numpy.random.uniform(-1, 1, (2, 3, 4, 3)).astype(
+            numpy.float32)
+
+    def check_im2col(self, ph, pw, gpu):
+        if gpu:
+            im2col = conv.im2col_gpu
+            img = cuda.to_gpu(self.img)
+        else:
+            im2col = conv.im2col_cpu
+            img = self.img
+
+        col = im2col(img, 1, 1, 1, 1, ph, pw)
```

Comment:
I have no idea about efficient way of testing them. A straight forward way is comparing with the result written with plain for loops as same as the current tests. It will be a bit complicated and need tests for itself, though I expect that the original implementation and this slow implementation act as testers for each other.